examples/huggingface/pytorch/text-generation/inference/run_generation_with_deepspeed.py

```diff
@@ -0,0 +1,578 @@
+
+parser.add_argument('-m', '--model-id',
+    type=str,
+    default='bigscience/bloom',
+    help="the huggingface mdoel id"
+)
+parser.add_argument('--device',
+    type=str,
+    choices=["cpu", "cuda", "xpu"],
+    help="cpu or cuda or xpu, same as --cuda or not",
+    default='xpu',
+)
+parser.add_argument(
+    "--dtype", type=str, help="float16 or bfloat16 or int8", choices=["int8", "float16", "bfloat16", "float32"], default="float16"
+)
+parser.add_argument("--local_rank", required=False, type=int, help="used by dist launchers")
+parser.add_argument("--batch_size", "--batch-size", default=1, type=int, help="batch size")
+parser.add_argument("--num-iter", default=10, type=int, help="num iter")
+parser.add_argument("--num-warmup", default=5, type=int, help="num warmup")
+parser.add_argument("--benchmark", action="store_true", help="additionally run benchmark")
+parser.add_argument("--cuda", action="store_true", help="run in cuda")
+parser.add_argument("--num-beams", type=int, nargs="+")
+parser.add_argument("--greedy", action="store_true")
+parser.add_argument("--ki", action="store_true")
+parser.add_argument('--max-new-tokens', default=32, type=int, nargs="+", help="output max new tokens")
+parser.add_argument('--input-tokens', default=32, type=str, nargs="+")
+parser.add_argument('--prompt', default=None, type=str)
+parser.add_argument('--ipex', action='store_true', help="ipex is not enabled now")
+parser.add_argument('--jit', action='store_true')
+parser.add_argument('--print-memory', action='store_true')
+parser.add_argument("--token-latency", action="store_true")
+parser.add_argument("--throughput", action="store_true")
+parser.add_argument("--accuracy-only", action="store_true")
+parser.add_argument(
+    "--acc-tasks",
+    nargs="+",
+    default=[
+        "lambada_standard",
+    ],
+    type=str,
+    help="tasks list for accuracy validation, only enabled lambada_standard and lambada_standard at present",
+)
+args = parser.parse_args()
+
+
+# import extension
+if args.ipex:
+    import intel_extension_for_pytorch as ipex
+    try:
+        ipex._C.disable_jit_linear_repack()
+    except Exception:
+        pass
+if args.jit:
+    torch._C._jit_set_texpr_fuser_enabled(False)
+
+if args.cuda or args.device == "cuda":
+    args.cuda = True
+    args.device == "cuda"
+
+def get_int_from_env(env_keys, default):
+    """Returns the first positive env value found in the `env_keys` list or the default."""
+    for e in env_keys:
+        val = int(os.environ.get(e, -1))
+        if val >= 0:
+            return val
+    return default
+
+local_rank = get_int_from_env(["LOCAL_RANK","MPI_LOCALRANKID","PALS_LOCAL_RANKID"], "0")
+world_size = get_int_from_env(["WORLD_SIZE","PMI_SIZE","PALS_LOCAL_SIZE"], "1")
+port = get_int_from_env(["MASTER_PORT"], 29500)
+print(f"*** local_rank={local_rank} world_size={world_size}")
+
+deepspeed.init_distributed(get_accelerator().communication_backend_name(), distributed_port=port)
+x = torch.ones(
```

Comment:
may I know what's it for? (line 125 to line 128)

Reply:
Is it a debug code?

Reply:
Hi, I added a comment for this. It's kept for testing ccl allreduce ahead of time.